src/Services/Azure/Authentication/CustomChainedCredential.cs

```diff
@@ -119,4 +135,76 @@ private static DefaultAzureCredential CreateDefaultCredential(string? tenantId)
 
         return new DefaultAzureCredential(defaultCredentialOptions);
     }
+
+    private static TokenCredential? CreateVsCodeBrokerCredential(string? tenantId)
+    {
+        const string vsCodeClientId = "aebc6443-996d-45c2-90f0-388ff96faa56";
+        string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
```

Comment:
user profile can be String.Empty on non-Windows operating systems here. Have you tested this?

Reply:
I have successfully tested on windows.
As per docs, User Profile returns $HOME on Mac OS and Linux.
https://learn.microsoft.com/en-us/dotnet/maui/macios/system-special-folders?view=net-maui-9.0
https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/8.0/getfolderpath-unix

I can double check for mac OS

Reply:
added the defensive check.